lib/pengine/pe_actions.c

```diff
@@ -507,15 +510,21 @@ validate_on_fail(const pe_resource_t *rsc, const char *action_name,
             }
 
             // Use value from first applicable promote action found
-            return on_fail;
+            key = strdup(XML_OP_ATTR_ON_FAIL);
+            new_value = strdup(promote_on_fail);
+            CRM_ASSERT((key != NULL) && (new_value != NULL));
+            g_hash_table_insert(meta, key, new_value);
+            return g_hash_table_lookup(meta, XML_OP_ATTR_ON_FAIL);
```

Comment:
Do we need a lookup when we just inserted it? I'd think `return new_value` would work, based on how `g_hash_table_insert()` handles a pre-existing key's value.

Reply:
Ah yes, that'll be in another PR